socketio/src/asynchronous/generator.rs

```diff
@@ -0,0 +1,34 @@
+use std::{pin::Pin, sync::Arc};
+
+use crate::error::Result;
+use futures_util::{ready, FutureExt, Stream, StreamExt};
+use tokio::sync::Mutex;
+
+pub(crate) type Generator<T> = Pin<Box<dyn Stream<Item = T> + 'static + Send>>;
```

Comment:
nit: a comment explaining why this is needed would be nice. Very much appreciate it being it's own type.